src/Microsoft.ComponentDetection.Contracts/TypedComponent/VcpkgComponent.cs

```diff
@@ -71,4 +81,22 @@ public override PackageURL PackageUrl
             }
         }
     }
+
+    private void SetGitRepoProperties()
+    {
+        /* example download locations
+         * "git+https://github.com/leethomason/tinyxml2@9.0.0"
+         * "git+https://github.com/Microsoft/vcpkg#ports/nlohmann-json"
+        */
+        var locationArr = this.DownloadLocation.Split('/');
+        if (!string.IsNullOrEmpty(locationArr[2]))
```

Comment:
Let's do an array length check, in which we expect at least 4 fragments, if less than that this should be noop. Otherwise we can hit index out of range exceptions here.